### DISCUSSION TITLE
This is a wsl2. Why the num_cores=6 while my cpu =12?

### DISCUSSION THREAD

**Discussion by tic-top** [modular/modular#911] [Q&A] [answered]
![image](https://github.com/modularml/mojo/assets/78676563/c0bc4259-0a83-4cf1-a874-d079b8a4f6a8)

**Comment by gujuriu** [comment, marked as answer]
Hi @tic-top, the [CPU](https://ark.intel.com/content/www/us/en/ark/products/196448/intel-core-i710710u-processor-12m-cache-up-to-4-70-ghz.html) from your picture has 6 cores (real ones) and two threads per core because of Intel® Hyper-Threading Technology / [HT](https://en.wikipedia.org/wiki/Hyper-threading) / [SMT](https://en.wikipedia.org/wiki/Simultaneous_multithreading), you've called a function name **num_cores()** and that function returned the right amount of CPU cores.

**Reply by tic-top** [reply]
Thank you so much!
